lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -50,9 +50,17 @@ defmodule Mix.Tasks.Escript.Install do
   end
 
   def before_install(_src, dst_path) do
-    File.rm(dst_path)
-    File.rm(dst_path <> ".bat")
-    :ok
+    case Mix.Utils.read_path(_src) do
```

Comment:
This will cause us to read the source twice. Given the source can be a network address, we don't want to do that. My suggestion: let's leave `before_install` as is and do the check in `after_install`, if we find out it is not an escript in `after_install`, we will remove the file and then raise an error with `Mix.raise`.